### ISSUE TITLE
placing water from bucket full game crash

### ISSUE THREAD

**Issue by NITRO1623** [anegostudios/VintageStory-Issues#4723]
### Game Version

v1.20 r4

### Platform

Windows

### Modded

Modded

### SP/MP

None

### Description

tried to place water for farm and game kicked completely back to desktop see below for critical log error

### How to reproduce

use bucket place water for farm

**Comment by NITRO1623** [comment]
Updating to v 1.20 pre 13 to see if update fixes crash

**Comment by Craluminum2413** [comment]
@NITRO1623 Does it happen without any mods?

**Comment by NITRO1623** [comment]
> @NITRO1623 Does it happen without any mods?

@Craluminum2413  yes just tried that as well

**Comment by radfast** [comment]
@NITRO1623  please can you post an extract from your `clientsettings.json` file found in the `%appdata%\VintagestoryData` folder. **Don't post the whole file as it contains some personal information like your email address!** But it will be very helpful to see these two settings, found in the intsettings section of the file:

    "maxTextureAtlasWidth": xxxx,
    "maxTextureAtlasHeight": xxxx,

**Comment by radfast** [comment]
Also, @NITRO1623:

1. How long are you usually playing for when the issue happens - like a few minutes, or many hours?
2. When it happened, were you in your base (or anyhow a player's house), or a story location, or in wilderness?
3. If it was your base or another player's house, please can you post a couple of screenshots, to give an idea of how much stuff is there and what types of machines and storage etc.  For example, a warehouse full of Crates might be relevant.  Armorstands might be relevant.

**Comment by NITRO1623** [comment]
> Also, @NITRO1623:
> 
> 1. How long are you usually playing for when the issue happens - like a few minutes, or many hours?
> 2. When it happened, were you in your base (or anyhow a player's house), or a story location, or in wilderness?
> 3. If it was your base or another player's house, please can you post a couple of screenshots, to give an idea of how much stuff is there and what types of machines and storage etc.  For example, a warehouse full of Crates might be relevant.  Armorstands might be relevant.



1. At that time of crash was probably playing for an hour

2. I was in my base I just found a new location from the better mods had a small village

3. Single player beginner stage had a few tools 3 chests and a wood bucket also had a lantern on the house.

**Comment by radfast** [comment]
Thank you. We'll try to reproduce, looking to get the same crash, I can see it was when the game attempted to place the full bucket of water on the ground  (I guess the wrong combination of Ctrl and Shift and click meant it wanted to place the bucket, not place the water block, happens to me all the time I can never remember which way round those are).

**Comment by radfast** [comment]
Any chance of those two settings from your `clientsettings.json`?  Even if you haven't touched them yourself, a mod might have

**Comment by NITRO1623** [comment]
> @NITRO1623  please can you post an extract from your `clientsettings.json` file found in the `%appdata%\VintagestoryData` folder. **Don't post the whole file as it contains some personal information like your email address!** But it will be very helpful to see these two settings, found in the intsettings section of the file:
> 
>     "maxTextureAtlasWidth": xxxx,
>     "maxTextureAtlasHeight": xxxx,

Thanks for the heads up I updated post I'll try to get you the file as requested

**Comment by radfast** [comment]
You've deleted the log with the crash report :(    Please re-post it, that's essential

**Comment by radfast** [comment]
```Game Version: v1.20.0-rc.4 (Unstable)
12/17/2024 1:27:36 PM: Critical error occurred
Loaded Mods: bettercrates@1.8.0-rc.1, driftershaveloot@1.0.0, primitivesurvival@3.7.2, game@1.20.0-rc.4, betterruins@0.4.1, carryon@1.8.0-pre.1, hudclockpatch@1.0.0, stepupcontinued@0.0.1, creative@1.20.0-rc.4, survival@1.20.0-rc.4
System.IndexOutOfRangeException: Index was outside the bounds of the array.
at Vintagestory.Client.NoObf.TextureAtlasManager.GetOrInsertTexture(AssetLocationAndSource loc, Int32& textureSubId, TextureAtlasPosition& texPos, CreateTextureDelegate onCreate, Single alphaTest)
at Vintagestory.GameContent.ContainerTextureSource.get_Item(String textureCode) in VSSurvivalMod\Systems\Liquid\ContainerTextureSource.cs:line 48
at Vintagestory.Client.NoObf.ShapeTesselator.TesselateShapeElement(Int32 indexForLogging, MeshData meshdata, ShapeElement element, TesselationMetaData meta) in VintagestoryLib\Client\Render\Tesselation\ShapeTesselator.cs:line 400
at Vintagestory.Client.NoObf.ShapeTesselator.TesselateShapeElements(MeshData meshdata, ShapeElement[] elements, TesselationMetaData meta) in VintagestoryLib\Client\Render\Tesselation\ShapeTesselator.cs:line 309
at Vintagestory.Client.NoObf.ShapeTesselator.TesselateShapeElements(MeshData meshdata, ShapeElement[] elements, TesselationMetaData meta) in VintagestoryLib\Client\Render\Tesselation\ShapeTesselator.cs:line 309
at Vintagestory.Client.NoObf.ShapeTesselator.TesselateShape(Shape shapeBase, MeshData& modeldata, Vec3f wholeMeshRotation, Vec3f wholeMeshOffset, Single wholeMeshScale, TesselationMetaData meta) in VintagestoryLib\Client\Render\Tesselation\ShapeTesselator.cs:line 250
at Vintagestory.GameContent.BlockLiquidContainerTopOpened.GenMesh(ICoreClientAPI capi, ItemStack contentStack, BlockPos forBlockPos) in VSSurvivalMod\Systems\Liquid\BlockLiquidContainerTopOpened.cs:line 153
at Vintagestory.GameContent.BlockEntityBucket.GenMesh() in VSSurvivalMod\BlockEntity\BEBucket.cs:line 62
at Vintagestory.GameContent.BlockEntityBucket.OnBlockPlaced(ItemStack byItemStack) in VSSurvivalMod\BlockEntity\BEBucket.cs:line 50```

**Comment by radfast** [comment]
In agreement with Craluminum2413 I think it's highly likely this issue is linked to #4693.  Specifically, if issue #4693 occurred just beforehand, then it would not have completed the previous allocation of space, causing this issue to occur on the next attempt.

As #4693 has been fixed in upcoming 1.20.0-rc.6, I'm called to call this one resolved as well.    (If it recurs in -rc.6 we can always re-open.)

@NITRO1623 there's nothing you can do your side except either: (1) play with fewer mods until 1.20-rc.6 is released; or (2) tolerate the occasional crash in -rc.5, hopefully not more than once a day.
